src/table/components/head/MenuList/RecursiveMenuList.tsx

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable import/no-cycle */
+import React from 'react';
+import Menu from '@mui/material/Menu';
+import { PopoverOrigin } from '@mui/material';
+import { MenuItemGroup } from '../../../types';
+import MenuGroupWrapper from './MenuGroupWrapper';
+
+interface RecursiveMenuListProps {
+  open: boolean;
+  anchorEl: HTMLDivElement | null;
+  onClose: () => void;
+  menuGroups: MenuItemGroup[];
+  transformOrigin?: PopoverOrigin; // eslint-disable-line react/require-default-props
+  anchorOrigin?: PopoverOrigin; // eslint-disable-line react/require-default-props
+  ariaLabel?: string; // eslint-disable-line react/require-default-props
+}
+
+const RecursiveMenuList = ({
+  anchorEl,
+  open,
+  onClose,
+  ariaLabel,
+  menuGroups,
+  transformOrigin,
+  anchorOrigin,
+}: RecursiveMenuListProps) => {
+  if (!menuGroups.length) return null;
+  return (
+    <Menu
+      className="sn-table-head-menu"
+      open={open}
+      anchorEl={anchorEl}
+      onClose={onClose}
+      autoFocus={false}
+      {...(ariaLabel ? { [`aria-labelledby`]: ariaLabel } : {})}
+      {...(anchorOrigin ? { anchorOrigin } : {})}
```

Comment:
Think these two can do the same as well?

Reply:
passing `undefined` (`anchorOrigin={anchorOrigin}`) causes below effect (MUI default anchor origin is top left of anchor ref):
https://user-images.githubusercontent.com/29652890/219617153-9ef6cc2d-d8a0-4a7f-b891-fe20e4e3e613.mov

So it will not work for popover related stuff but it does for `aria-labeledby`, will update it asap

Reply:
So not passing in anchorOrigin vs anchorOrigin=undefined gives different results. I'd say I did not expect that

Reply:
Yes, passing undefined fall into default behaviour of mui which we dont want